src/core/contracts/identity.ts

```diff
@@ -377,45 +380,95 @@ export const removeDelegate = async (
 };
 
 /**
- * getAddDelegateLogs() Retrieves event filter for DSNPAddDelegate event
+ * getDelegeteIdentitiesFor() Retrieves all active identities for an address
  *
- * @param ethereumAddress - Address to filter on
+ * @param ethereumAddress - The address to find all active identities associated to it
  * @param opts - Optional. Configuration overrides, such as from address, if any
- * @returns DSNPBatch event filter
  */
-export const getAddDelegateLogData = async (
+export const getDelegeteIdentitiesFor = async (
   ethereumAddress: EthereumAddress,
   opts?: ConfigOpts
-): Promise<DelegateAddLogData[]> => {
-  const eventSignature = hash("DSNPAddDelegate(address,uint8)");
-  const addDelegateLogs = await getDelegateLogs(ethereumAddress, eventSignature, opts);
-  const addDelegateLogData: DelegateAddLogData[] = siftDelegateLogs(addDelegateLogs);
+): Promise<EthereumAddress[]> => {
+  const delegateLogData: DelegateLogData[] = siftDelegateLogs(await getAllDelegateLogsFor(ethereumAddress, opts));
+
+  const activeIdentities = (await resolveDelegatesFor(ethereumAddress, delegateLogData)).map(
+    (data: DelegateLogData) => data?.identityAddress
+  );
 
-  return addDelegateLogData;
+  return activeIdentities;
 };
 
 /**
- * getRemoveDelegateLogs() Retrieves event filter for DSNPRemoveDelegate event
+ * resolveDelegatesFor() Filter delegates that have not been removed
  *
- * @param ethereumAddress - Address to filter on
+ * @param ethereumAddress - The address to find all active identities associated to it
+ * @param delegateLogData - Log data for delegates (DSNPAddDelegate and DSNPRemoveDelegate)
```

Comment:
If I understand this correctly, this function expects unfiltered log data from the Add/Remove Delegate events, is that right?